src/smart-components/role/add-role/add-role-wizard.js

```diff
@@ -105,37 +107,58 @@ const AddRoleWizard = ({ pagination, filters, orderBy }) => {
       'role-copy-name': copyName,
       'role-copy-description': copyDescription,
       'add-permissions-table': permissions,
+      'inventory-groups-role': invGroupsRole,
       'cost-resources': resourceDefinitions,
       'role-type': type,
     } = formData;
-    setWizardContextValue((prev) => ({ ...prev, submitting: true }));
-
     const selectedPermissionIds = permissions.map((record) => record.uuid);
+
+    const resourceDefinitionsMap = {};
+    resourceDefinitions?.forEach((r) => {
+      resourceDefinitionsMap[r.permission] = r.resources;
+    });
```

Comment:
No need to assign, it's faster to use `reduce`
```suggestion
    const resourceDefinitionsMap = resourceDefinitions?.reduce((acc, { permission, resources }) => ({
       ...acc,
       [permission]: resources
    }), {}) || {};
```